### DISCUSSION TITLE
Centering a window with open space

### DISCUSSION THREAD

**Discussion by WJCFerguson** [i3/i3#4547] [Q&A]
When I have one single window in a workspace, I often want it to be narrower than the whole width - especially the browser.  I have a fairly fragile solution below, but I was wondering if there are other better ways to achieve it.

My bindings create `open` space on either side with one key, and then kills one window to left and right with another.  If I'm careful, it works for me, but obviously it could be more robust.

```
# when you have one window, center it with open space on either side:
bindsym <KEY> split h, open, open, focus left, move left, resize shrink width 15 px or 15 ppt, focus right, focus right, resize shrink width 15 px or 15 ppt, focus left
# crude undo by killing to the left and right.
bindsym Shift+<KEY> focus right, kill, focus left, kill
```

I wish `open` was a more first class citizen of i3.  The most obvious minor improvement would be having the second binding only kill the windows to each side if they're `open`(?).

I'm using i3-gaps by the way.  I don't _think_ there's a way for me to use that to create huge side margins for the window rather than `open` space(?).  That would probably be better if it were possible.

**Comment by Airblader** [comment]
Please note that `open` is an internal command that we do not support.

Your two options would be to float the browser, set the size and center it, or to use i3-gaps and use large outer gaps.
